### ISSUE TITLE
Missrendering while playing 10bit conent with hwdec (v4l2m2m) in X11

### ISSUE THREAD

**Issue by rox4d** [mpv-player/mpv#8552]
### Important Information
- mpv version - current git.
- Linux Distribution and Version - ubuntu-18.04 arm64 on rockpro64 sbc (rk3399 based board),mesa from git, kernel 5.10.12 with patches from openelec.
- Source of the mpv binary - compiled from git.
- If known which version of mpv introduced the problem - it is unknown.
- Window Manager and version - marco-compton.
- GPU driver and version - panfrost from mesa current git.
- Possible screenshot or video of visual glitches
![mpv-shot0001](https://user-images.githubusercontent.com/34712490/107659947-c7eda080-6c98-11eb-9511-2a4c25cb4aeb.jpg)
 
### Reproduction steps
mpv --no-config -vo=gpu -gpu-context=x11egl -hwdec jellyfish-20-mbps-hd-hevc-10bit.mkv
(there are no issue without hwdec, besides performance),
This movie also plays fine in drm console with:
mpv --no-config -vo=gpu -gpu-context=drm -hwdec --drm-draw-plane=overlay --drm-drmprime-video-plane=primary jellyfish-20-mbps-hd-hevc-10bit.mkv
(mpv use FFmpeg with Kwiboo's patches for v4l2-request and rkvdec)
### Expected behavior
Correct rendering.
### Actual behavior
Missrendering
### Log file
[mpv_log_x11.txt](https://github.com/mpv-player/mpv/files/5966501/mpv_log_x11.txt)
### Sample files
https://jell.yfish.us/media/jellyfish-20-mbps-hd-hevc-10bit.mkv
### 
I am completely not sure that issue is on mpvs side, how determine it?

**Comment by haasn** [comment]
```
[   0.324][i][vd] Using hardware decoding (drm-copy).
[   0.324][v][vd] Decoder format: 1920x1080 nv12 auto/auto/auto/limited/auto CL=unknown
```

Seems odd, since nv12 is an 8-bit format. Does anybody know if this hwdec is doing conversion internally? (If so, that's likely where the bug is)

Also, for some reason the mpv man page does not reference the `drm` hwdec, what's up with that?

**Comment by rox4d** [comment]
> Seems odd, since nv12 is an 8-bit format.

Yes, but in case of usage mpv with gpu-context=drm (which is works) it's the same:

 [   0.340][i][vd] Using hardware decoding (drm).
 [   0.341][v][vd] Decoder format: 1920x1080 drm_prime[nv12] auto/auto/auto/limited/auto CL=unk

(https://github.com/mpv-player/mpv/files/5953570/mpv_log.txt)

**Comment by rox4d** [comment]
OS was replaced, now it is Debian “bookworm” with linux-6.0.10 + patches from LibreELEC (https://github.com/LibreELEC/LibreELEC.tv/tree/master/projects/Rockchip/patches/linux/default) and ffmpeg-5.1.2 with v4l2-request patches (https://github.com/jernejsk/FFmpeg/tree/v4l2-request-n5.1.2) compiled with modified v4l2-controls.h, videodev2.h from kernel source (linux-6.0.10/include/uapi/linux/) and placed in /usr/include/linux/
Now mpv --no-config -vo=gpu -gpu-context=x11egl -hwdec jellyfish-20-mbps-hd-hevc-10bit.mkv
show blue window, with such messages in console:
~~~
 (+) Video --vid=1 (*) (hevc 1920x1080 29.970fps)
Cannot load libcuda.so.1
[ffmpeg] AVHWDeviceContext: Cannot load libcuda.so.1
[ffmpeg] AVHWDeviceContext: Could not dynamically load CUDA
Failed to open VDPAU backend libvdpau_rockchip.so: cannot open shared object file: No such file or directory
[ffmpeg/video] hevc: v4l2_request_probe_video_device: try output format failed
[ffmpeg/video] hevc: v4l2_request_probe_video_device: try output format failed
[ffmpeg/video] hevc: ff_v4l2_request_query_control: query control failed, Invalid argument (22)
Using hardware decoding (drm).
VO: [gpu] 1920x1080 drm_prime[nv12]
[vo/gpu/drmprime] Cannot map unknown multi-plane format: 0x3531564E
[vo/gpu/drmprime] mapping DRM dmabuf failed
[vo/gpu] Mapping hardware decoded surface failed.
V: 00:00:00 / 00:00:30 (0%)
[vo/gpu/drmprime] Cannot map unknown multi-plane format: 0x3531564E
[vo/gpu/drmprime] mapping DRM dmabuf failed
[vo/gpu] Mapping hardware decoded surface failed.
V: 00:00:00 / 00:00:30 (0%)
~~~
google say that 0x3531564E are NV15 little-endian  format, It works fine in drm console (but nv12?):
~~~
$ mpv --no-config -drm-draw-plane=overlay -drm-drmprime-video-plane=primary -hwdec jellyfish-20-mbps-hd-hevc-10bit.mkv
 (+) Video --vid=1 (*) (hevc 1920x1080 29.970fps)
[vo/gpu] VT_GETMODE failed: Inappropriate ioctl for device
[vo/gpu/opengl] Failed to set up VT switcher. Terminal switching will be unavailable.
Cannot load libcuda.so.1
[ffmpeg] AVHWDeviceContext: Cannot load libcuda.so.1
[ffmpeg] AVHWDeviceContext: Could not dynamically load CUDA
[ffmpeg/video] hevc: v4l2_request_probe_video_device: try output format failed
[ffmpeg/video] hevc: v4l2_request_probe_video_device: try output format failed
[ffmpeg/video] hevc: ff_v4l2_request_query_control: query control failed, Invalid argument (22)
Using hardware decoding (drm).
VO: [gpu] 1920x1080 drm_prime[nv12]
[vo/gpu] Using HW-overlay mode. No GL filtering is performed on the video!
V: 00:00:01 / 00:00:30 (3%)

Exiting... (Quit)
~~~

**Comment by rox4d** [comment]
kodi also can't show this (via EGL, Direct to Plane is fine)
